### ISSUE TITLE
Does it stores temporary data of filesystem?

### ISSUE THREAD

**Issue by fdistorted** [r3nt0n/bopscrk#12]
I have started the command with 26 words, 4 words combinations and all the transformations.
It has been running for 8 days already but looks like it hasn't written anything to FS yet. Is it OK?

Below is everything that I got as output on the first day.
Thanks.
![image](https://user-images.githubusercontent.com/20381690/102187473-b23c4480-3ebc-11eb-9438-065f636f789c.png)

**Comment by r3nt0n** [comment]
No, output file is written at the end, but it could be a great improvement to introduce in next version.

Thanks!

**Comment by fdistorted** [comment]
Thanks!
Great. That also would be great to track progress somehow.
It is running for for 11 days already on my mini server. 
A few times a year I have power power outage and I hope it won't happen while it doing a password list :)

I have one more question: is password list size limited with ram size?

**Comment by r3nt0n** [comment]
Yes @fdistorted , right now the final wordlists are limited to the RAM size of your system. This project started as a focused wordlist generator, so I didn't take care about manage wordlists larger than 3-4 GB. In future commits I will try to change this behaviour, saving the words into the output file and reading from it to apply transforms, again saving new words as soon as it is produced.

**Comment by fdistorted** [comment]
I think that would be great to add session parameter to store all save files under the session name. To continue user should use session parameter again. As I see the hardest part is to track and store the progress on the parallel executions. Maybe everything should be divided by chunks and the last calculated chunk number should be stored after it's completion.

**Comment by r3nt0n** [comment]
Hello @amanenk I've been out of this project for a while, recently I've tried to update it including this memory management improve, still is not working, anyway if you want to check out my scratch please see the new function I wrote `transform_cached_wordlist_and_save` inside `modules/transforms.py` and let me know what you think.

Thank you :)

**Comment by its0x08** [comment]
> Yes @fdistorted , right now the final wordlists are limited to the RAM size of your system. This project started as a focused wordlist generator, so I didn't take care about manage wordlists larger than 3-4 GB. In future commits I will try to change this behaviour, saving the words into the output file and reading from it to apply transforms, again saving new words as soon as it is produced.

Try better 70-80% of the RAM because if you use it all, it will leave no RAM for other Apps and OS so the PC will lag badly( as with other tools I have seen)

**Comment by avirils** [comment]
I started a command with 90 words,5  words combination.
Instead of it to go directly the combination it first make a
Combination of 90 words using 2 words
Combination of 90 words using 3 words
Combination of 90 words using 4 words
Before it's start making the
combination of 90 words using 5 words
And it consumes a lot of time,is they any way I can edit the "bopscrk.cfg" file or any command to stop it?  @r3nt0n 
THANKS
